[DOCS] Documents trained model auto-scaling (#2795)
elastic/stack-docs: docs/en/stack/ml/nlp/ml-nlp-deploy-models.asciidoc

When you have dedicated deployments for different purposes, you ensure that the search speed remains unaffected by ingest workloads, and vice versa.
Having separate deployments for search and ingest mitigates performance issues resulting from interactions between the two, which can be hard to diagnose.

[role="screenshot"]
image::images/ml-nlp-deployment-id-elser-v2.png["Model deployment on the Trained Models UI."]

Each deployment will be fine-tuned automatically based on its specific purpose you choose.

NOTE: Since eland uses APIs to deploy the models, you cannot see the models in {kib} until the saved objects are synchronized.
You can follow the prompts in {kib}, wait for automatic synchronization, or use the {kibana-ref}/machine-learning-api-sync.html[sync {ml} saved objects API].

You can define the resource usage level of the NLP model during model deployment.
The resource usage levels behave differently depending on <<nlp-model-adaptive-resources, adaptive resources>> being enabled or disabled.
When adaptive resources are disabled but {ml} autoscaling is enabled, vCPU usage of Cloud deployments derived from the Cloud console and functions as follows:

> * Low: This level limits resources to one vCPU, which may be suitable for development, testing, and demos depending on your parameters.

Reviewer: We set 2 vCPUs for low. 
```suggestion
* Low: This level limits resources to two vCPUs, which may be suitable for development, testing, and demos depending on your parameters.
```

It is not recommended for production use

> * Medium: This level limits resources to two vCPUs, which may be suitable for development, testing, and demos depending on your parameters.

Reviewer: 32 for medium 

```suggestion
* Medium: This level limits resources to 32 vCPUs, which may be suitable for development, testing, and demos depending on your parameters.
```

It is not recommended for production use.

> * High: This level may use the maximum number of vCPUs available for this deployment from the Cloud console.

Reviewer: As we do not have access to vCPUs ranges at the moment, current value for high static allocations is 128. cc @arisonl 

```suggestion
* High: This level may use the maximum number of vCPUs available for this deployment from the Cloud console.
```
